### PR TITLE
source selector parent, item and create component

### PR DESCRIPTION
**Pull request by AnkRaiza**
relates #578 #583 

image

image

image

image

### CODE REVIEW

File: src/components/SourceSelector/SourceSelectorCreateComponent.test.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import * as chai from "chai";
+import {shallow} from "enzyme";
+import * as React from "react";
+import {IconButton} from "react-toolbox";
+
+import { Source } from "../../models/source";
+import { dummySources } from "../../utils/test";
+import SourceSelectorCreate from "./SourceSelectorCreateComponent";
+
+let expect = chai.expect;
```
Comment (jperata) [inline src/components/SourceSelector/SourceSelectorCreateComponent.test.tsx]: const

File: src/components/SourceSelector/SourceSelectorItemComponent.test.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import {shallow} from "enzyme";
+import * as React from "react";
+import {IconButton} from "react-toolbox";
+import Dialog from "react-toolbox/lib/dialog";
+
+import { Source } from "../../models/source";
+import { dummySources } from "../../utils/test";
+import SourceSelectorItem from "./SourceSelectorItemComponent";
+
+let expect = chai.expect;
```
Comment (jperata) [inline src/components/SourceSelector/SourceSelectorItemComponent.test.tsx]: const

File: src/components/SourceSelector/SourceSelectorItemComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        this.setState((prevState) => ({
+            ...prevState,
+            deleteDialogActive: !prevState.deleteDialogActive,
+        }));
+    }
+
+    async handleEnableValidation (e: any) {
+        e.stopPropagation();
+        const validation_enabled = !this.state.enableValidation;
+        console.log(validation_enabled);
```
Comment (jperata) [inline src/components/SourceSelector/SourceSelectorItemComponent.tsx]: remove

File: src/components/SourceSelector/SourceSelectorItemComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+                    }
+                    <IconButton className={SourceSelectorItemStyle.delete_button} icon={"delete"} onClick={this.handleDeleteDialogToggle}/>
+                    <Dialog
+                        theme={DeleteDialogTheme}
+                        actions={this.dialogActions}
+                        active={this.state.deleteDialogActive}
+                        onEscKeyDown={this.handleDeleteDialogToggle}
+                        onOverlayClick={this.handleDeleteDialogToggle}
+                        title="Delete Source" >
+                        <p>Are you sure you want to delete {this.props.source.name}? This action can not be undone.</p>
```
Comment (jperata) [inline src/components/SourceSelector/SourceSelectorItemComponent.tsx]: missing $ in {this.props.source.name}

File: src/components/SourceSelector/SourceSelectorParentComponent.test.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import * as sinon from "sinon";
+
+import { Source } from "../../models/source";
+import auth from "../../services/auth";
+import { dummySources } from "../../utils/test";
+import SourceSelectorCreate from "./SourceSelectorCreateComponent";
+import SourceSelectorItem from "./SourceSelectorItemComponent";
+import { SourceSelector } from "./SourceSelectorParentComponent";
+
+let expect = chai.expect;
```
Comment (jperata) [inline src/components/SourceSelector/SourceSelectorParentComponent.test.tsx]: const

File: src/components/SourceSelector/SourceSelectorItemComponent.tsx (hunk 4 above)
Comment (AnkRaiza) [reply src/components/SourceSelector/SourceSelectorItemComponent.tsx]: this is not a `` is just a {} inside an html tag i don't think it needs the $

Comment (jperata) [reply src/components/SourceSelector/SourceSelectorItemComponent.tsx]: :+1:

**Pull request merged**